pkg/controller/operators/suite_test.go

```diff
@@ -88,14 +88,14 @@ var _ = BeforeSuite(func() {
 	useExisting := false
 	testEnv = &envtest.Environment{
 		UseExistingCluster: &useExisting,
-		CRDs: []apiextensionsv1.CustomResourceDefinition{
-			*crds.CatalogSource(),
-			*crds.ClusterServiceVersion(),
-			*crds.InstallPlan(),
-			*crds.Subscription(),
-			*crds.OperatorGroup(),
-			*crds.Operator(),
-			*crds.OperatorCondition(),
+		CRDs: []*apiextensionsv1.CustomResourceDefinition{
```

Comment:
Interesting - it looks like this was changed in https://github.com/kubernetes-sigs/controller-runtime/commit/a14a68cdaba04ccbc2dd4501f162f8472f11d800 but if I'm reading that commit tagging right, it's only available in v0.11.0 versions of c-r but we still replace pin to my fork's v0.10.x version. Any idea on why we need to make this change? In order to make CI happy as we're pulling in the latest envtest binary now, so there's some potential skew between our c-r version and entest?

Reply:
I may have missed something:
~~~
require (
[...]
sigs.k8s.io/controller-runtime v0.11.0
[...]
replace (
sigs.k8s.io/controller-runtime v0.10.0 => github.com/timflannagan/controller-runtime v0.10.1-0.20211210161403-6756a4203e70
~~~
The replace version v0.10.0 does not match the request version v0.11.0. This means that v0.11.0 was in use. With your version of controller-runtime there are incompatibilities with the logr version brought by o-f/api, for instance:
~~~
vendor/sigs.k8s.io/controller-runtime/pkg/log/deleg.go:79:12: undefined: logr.WithCallDepth
vendor/sigs.k8s.io/controller-runtime/pkg/log/deleg.go:163:2: cannot use res (type *DelegatingLogger) as type logr.Logger in return argument
~~~

I am looking at: #2353

Reply:
I looked at #2368 but did not see why you decided to keep being on a fork. [The issue](https://github.com/kubernetes-sigs/controller-runtime/pull/1650) you mentioned in regards to the way metadata is handled in controller-runtime has been merged with commit 74ba5d77bf99d8968543f3536e4d49a2e3818fee is in [v0.11](https://github.com/kubernetes-sigs/controller-runtime/pull/1650/commits/74ba5d77bf99d8968543f3536e4d49a2e3818fee)
Am I missing something?

Reply:
I'm somewhat blanking on the overall context as to why that fork was necessary when bumping the c-r dependency in OLM to v0.10.x at the time. I remember seeing e2e was constantly failing despite the changes being fairly innocuous, and we were under the impression that Joe's upstream fix solved the issue we originally had to hack around. Let me see if I can dig up any more context today, but given the current e2e checks are passing, it's possible we had misdiagnosed the root case at the time.